internal/manifests/collector/rbac.go

```diff
@@ -18,22 +18,33 @@ import (
 	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	"github.com/open-telemetry/opentelemetry-operator/internal/api/convert"
 	"github.com/open-telemetry/opentelemetry-operator/internal/manifests"
 	"github.com/open-telemetry/opentelemetry-operator/internal/manifests/collector/adapters"
 	"github.com/open-telemetry/opentelemetry-operator/internal/manifests/manifestutils"
 	"github.com/open-telemetry/opentelemetry-operator/internal/naming"
 )
 
-func ClusterRole(params manifests.Params) *rbacv1.ClusterRole {
-	configFromString, err := adapters.ConfigFromString(params.OtelCol.Spec.Config)
+func ClusterRole(params manifests.Params) (*rbacv1.ClusterRole, error) {
+	otelCol, err := convert.V1Alpha1to2(params.OtelCol)
+	if err != nil {
+		return nil, err
+	}
+
+	confStr, err := otelCol.Spec.Config.Yaml()
```

Comment:
The typed config is serialized to yaml and then back unserialized to  `map[interface{}]interface{}`

The typed config could be easily converted to `map[interface{}]interface{}` which is accepted in most signatures

Reply:
please book a follow up ticket for this

Reply:
https://github.com/open-telemetry/opentelemetry-operator/issues/2603